File: DAL/Entities/DoctorSchedule.cs
using DAL.Entities.Entity;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace DAL.Entities
{
    public enum Days
    {
        Monday,
        Tuesday,
        Wednesday,
        Thursday,
        Friday,
        Saturday,
        Sunday
    }

    public class DoctorSchedule : Entity<int>
    {
        [ForeignKey("Doctor")]
        public int DoctorId { get; set; }
        public Doctor Doctor { get; set; }
        [EnumDataType(typeof(Days))]
        public Days Day { get; set; }
        public TimeSpan StartTime { get; set; }
        public TimeSpan EndTime { get; set; }
    }
}

File: PL/ClientApp/src/app/_models/doctorSchedule.ts
import { TimeSpan } from "./timespan";

export const Days: string[] = [
  "Monday",
  "Tuesday",
  "Wednesday",
  "Thursday",
  "Friday",
  "Saturday",
  "Sunday"
];

export class DoctorSchedule {
  id: number;
  doctorId: number;
  day: number;
  startTime: string;
  endTime: string;
}
